src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -543,54 +542,60 @@ protected void onShutdown() throws InterruptedException {
+      throw new ConfigurationException("worker's public name should not be empty");
     }
 
     return builder.build();
   }
 
   private static void printUsage(OptionsParser parser) {
     logger.log(INFO, "Usage: CONFIG_PATH");
-    logger.log(INFO,
-        parser.describeOptions(
-            Collections.emptyMap(), OptionsParser.HelpVerbosity.LONG));
+    logger.log(
+        INFO, parser.describeOptions(Collections.emptyMap(), OptionsParser.HelpVerbosity.LONG));
   }
 
   public static void main(String[] args) throws Exception {
-    // Only log severe log messages from Netty. Otherwise it logs warnings that look like this:
-    //
-    // 170714 08:16:28.552:WT 18 [io.grpc.netty.NettyServerHandler.onStreamError] Stream Error
-    // io.netty.handler.codec.http2.Http2Exception$StreamException: Received DATA frame for an
-    // unknown stream 11369
-    nettyLogger.setLevel(SEVERE);
-
-    OptionsParser parser = OptionsParser.newOptionsParser(WorkerOptions.class);
-    parser.parseAndExitUponError(args);
-    List<String> residue = parser.getResidue();
-    if (residue.isEmpty()) {
-      printUsage(parser);
-      throw new IllegalArgumentException("Missing CONFIG_PATH");
-    }
-    Path configPath = Paths.get(residue.get(0));
-    String session = UUID.randomUUID().toString();
-    Worker worker;
-    try (InputStream configInputStream = Files.newInputStream(configPath)) {
-      worker =
-          new Worker(
-              session,
-              toShardWorkerConfig(
-                  new InputStreamReader(configInputStream),
-                  parser.getOptions(WorkerOptions.class)));
+
+    try {
+      // Only log severe log messages from Netty. Otherwise it logs warnings that look like this:
+      //
+      // 170714 08:16:28.552:WT 18 [io.grpc.netty.NettyServerHandler.onStreamError] Stream Error
+      // io.netty.handler.codec.http2.Http2Exception$StreamException: Received DATA frame for an
+      // unknown stream 11369
+      nettyLogger.setLevel(SEVERE);
+
+      OptionsParser parser = OptionsParser.newOptionsParser(WorkerOptions.class);
+      parser.parseAndExitUponError(args);
+      List<String> residue = parser.getResidue();
+      if (residue.isEmpty()) {
+        printUsage(parser);
+        throw new IllegalArgumentException("Missing CONFIG_PATH");
+      }
+      Path configPath = Paths.get(residue.get(0));
+      String session = UUID.randomUUID().toString();
+      Worker worker;
+      try (InputStream configInputStream = Files.newInputStream(configPath)) {
+        worker =
+            new Worker(
+                session,
+                toShardWorkerConfig(
+                    new InputStreamReader(configInputStream),
+                    parser.getOptions(WorkerOptions.class)));
+      }
+      worker.start();
+      worker.blockUntilShutdown();
+      System.exit(0); // bullet to the head in case anything is stuck
+    } catch (Exception e) {
+      Worker.logger.log(SEVERE, e.toString());
```

Comment:
I would prefer not to fully indent a method with a try block - prefer a separate method call in this case, and there are some other issues:
There's no need to user the `Worker.` here, you're in a static method.
The logging message should not be the string rendering of the exception. `logger.log(SEVERE, "exception caught", e)` should suffice here
I'd recommend `System.exit(1)`, or to rethrow the exception (probably the former, given that you're logging explicitly).